force-app/main/default/classes/FSL/WorkOrderImportController.cls

```diff
@@ -0,0 +1,456 @@
+            Integer i = 0;
+            for (Database.SaveResult sr : saveResultWorkOrderList) {
+                i++;
+                if (!sr.isSuccess()) {
+                    ErrorEntity e = new ErrorEntity('1',
+                                                    String.valueOf(i),
+                                                    new List<String>
+                                                    {sr.getErrors()[0].getMessage()});
+                    errorsToFrontList.add(e);
+                    flag = true;
+                }
+            }
+            if (flag) {
+                Database.rollback( sp );
+            }
+        }
+        catch (Exception ex) {
+            throw new AuraHandledException(ex.getMessage());
+        }
+        if (checkErrorsToFront(errorsToFrontList)) {
+            return null;
+        }
+        Map<Id, Id> workOrderIdMap = new Map<Id, Id>();
+        for (WorkOrder w : workOrders) {
+            workOrderIdMap.put(w.PropertyId__c, w.Id);
+        }
+        List<Id> workOrderExcelAmountIds = new List<Id>();
+        for (WorkOrderImportModel.WorkOrderExcel w : workOrderExcelList) {
+            workOrderExcelAmountIds.add(workOrderIdMap.get(w.NewPropertyId));
+        }
+        return workOrderExcelAmountIds;
+    }
+
+    private static List<Id> createWorkOrderLineItems (
+        List<WorkOrderImportModel.WorkOrderLineItemExcel> workOrderLineItemExcelList,
+        List<Id> dbWorkOrderIds
+        ) {
+        List<WorkOrderLineItem> workOrderLineItems = new List<WorkOrderLineItem>();
+        for (WorkOrderImportModel.WorkOrderLineItemExcel wOrderLineItemExcel :
+             workOrderLineItemExcelList) {
+            try {
+                WorkOrderLineItem wOrderLineItem = new WorkOrderLineItem(
+                    WorkOrderId = dbWorkOrderIds.get(Integer.valueOf(wOrderLineItemExcel.FakeId) - 1),
+                    Category__c = wOrderLineItemExcel.Category,
+                    SubCategory__c = wOrderLineItemExcel.SubCategory,
+                    Issue__c = wOrderLineItemExcel.Issue,
+                    Description = wOrderLineItemExcel.ProblemDescription,
+                    Yardi_Work_Order_Number__c = Integer.valueOf(wOrderLineItemExcel.YardiWorkOrderNumber)
+                    );
+                workOrderLineItems.add(wOrderLineItem);
+            }
+            catch (Exception ex) {
+                throw new AuraHandledException(
+                          String.format(
+                              '"{0}" related to line number {1}',
+                              new List<String> { ex.getMessage(), wOrderLineItemExcel.FakeId }
+                              )
+                          );
+            }
+        }
+        Boolean flag = false;
+        try {
+            Database.SaveResult[] saveResultList = Database.insert(workOrderLineItems, false);
+            Integer i = 0;
```

Comment:
Looks like not used

Reply:
**flag** checks if it least 1 error occured, **i** counts how many records were saved